src/AuthorizationClient.js

```diff
@@ -1159,6 +1159,120 @@ class AuthorizationClient {
+   * @param objectId
+   * @returns {Promise<string|undefined>}
+   */
+  async GetTenantContractId(libraryId, objectId) {
+    const hasGetMetaMethod = await this.ContractHasMethod({
+      contractAddress: Utils.HashToAddress(objectId),
+      methodName: "getMeta"
+    });
+
+    let tenantContractId;
+    if(hasGetMetaMethod) {
+      tenantContractId = await this.client.CallContractMethod({
+        contractAddress: Utils.HashToAddress(objectId),
+        methodName: "getMeta",
+        methodArgs: [
+          "_ELV_TENANT_ID"
+        ]
+      });
+      if(tenantContractId !== "") {
+        return Buffer.from(tenantContractId.replace("0x", ""), "hex").toString("utf8");
+      }
+    }
+
+    return await this.client.ContentObjectMetadata({
+      libraryId,
+      objectId,
+      metadataSubtree: "tenantContractId",
+    });
+  }
+
+  /**
+   * SetTenantContractId sets the ID of the tenant contract for the
+   * objectId provided in contract metadata or in fabric metadata
+   *
+   * @param libraryId
+   * @param objectId
+   * @param tenantContractId
+   * @returns {Promise<{tenantId: (undefined|string), tenantContractId: string}>}
+   */
+  async SetTenantContractId(libraryId, objectId, tenantContractId) {
+    if(tenantContractId && (!tenantContractId.startsWith("iten") || !Utils.ValidHash(tenantContractId))) {
+      throw Error(`Invalid tenant ID: ${tenantContractId}`);
+    }
+    const tenantAddress = Utils.HashToAddress(tenantContractId);
+
+    const version = await this.client.AccessType({id: tenantContractId});
+    if(version !== this.ACCESS_TYPES.TENANT) {
+      throw Error("Invalid tenant ID: " + tenantContractId);
+    }
+
+    // get tenant admin group
+    const tenantAdminGroupAddress = await this.client.CallContractMethod({
+      contractAddress: tenantAddress,
+      methodName: "groupsMapping",
+      methodArgs: ["tenant_admin", 0],
+      formatArguments: true,
+    });
+
+    const hasPutMetaMethod = await this.ContractHasMethod({
+      contractAddress: Utils.HashToAddress(objectId),
+      methodName: "putMeta"
+    });
+
+    if(hasPutMetaMethod) {
+      await this.client.CallContractMethodAndWait({
+        contractAddress: Utils.HashToAddress(objectId),
+        methodName: "putMeta",
+        methodArgs: [
+          "_ELV_TENANT_ID",
+          tenantContractId
+        ]
+      });
+    } else {
+      await this.MergeObjectMetadata({
+        libraryId,
+        objectId,
+        metadata: {
+          tenantContractId,
+          tenantId: !tenantAdminGroupAddress ? undefined : `iten${Utils.AddressToHash(tenantAdminGroupAddress)}`
+        },
+        commitMessage: "set tenant_contract_id"
+      });
+    }
+    return {
+      tenantContractId: tenantContractId,
+      tenantId: !tenantAdminGroupAddress ? undefined : `iten${Utils.AddressToHash(tenantAdminGroupAddress)}`
+    };
+  }
+
+  async MergeObjectMetadata({libraryId, objectId, metadataSubtree = "/", metadata = {}, commitMessage}) {
```

Comment:
This method is unnecessary, especially since it's only used in one place, just do this properly where it's called

If you want it to be more concise, you can use client.EditAndFinalizeContentObject